mindsdb/integrations/utilities/rag/loaders/vector_store_loader/vector_store_loader.py

```diff
@@ -0,0 +1,117 @@
+
+    class Config:
+        arbitrary_types_allowed = True
+        extra = "forbid"
+        validate_assignment = True
+
+    def load(self) -> VectorStore:
+        """
+        Loads the vector store based on the provided config and embeddings model
+        :return:
+        """
+        self.vector_store = VectorStoreFactory.create(self.embeddings_model, self.config)
+        return self.vector_store
+
+
+class VectorStoreFactory:
+    @staticmethod
+    def create(embeddings_model: Embeddings, config: dict):
+
+        settings = VectorStoreConfig(**config)
+
+        if settings.vector_store_type == VectorStoreType.CHROMA:
+            return VectorStoreFactory._load_chromadb_store(embeddings_model, settings)
+        elif settings.vector_store_type == VectorStoreType.PGVECTOR:
+            return VectorStoreFactory._load_pgvector_store(embeddings_model, settings)
+        else:
+            raise ValueError(f"Invalid vector store type, must be one either {VectorStoreType.__members__.keys()}")
+
+    @staticmethod
+    def _load_chromadb_store(embeddings_model: Embeddings, settings) -> Chroma:
+        return Chroma(
+            persist_directory=settings.persist_directory,
+            collection_name=settings.collection_name,
+            embedding_function=embeddings_model,
+        )
+
+    @staticmethod
+    def _load_pgvector_store(embeddings_model: Embeddings, settings) -> PGVector:
+        # create an empty store if collection_name does not exist otherwise load the existing collection
+        store = PGVector(
+            connection_string=settings.connection_string,
+            collection_name=settings.collection_name,
+            embedding_function=embeddings_model
+        )
+        return VectorStoreFactory._load_data_into_langchain_pgvector(settings, store)
+
+    @staticmethod
+    def _load_data_into_langchain_pgvector(settings, vectorstore: PGVector) -> PGVector:
+        """
+        Fetches data from the existing pgvector table and loads it into the langchain pgvector vector store
+        :param settings:
+        :param vectorstore:
+        :return:
+        """
+        df = VectorStoreFactory._fetch_data_from_db(settings)
+
+        df[TableField.EMBEDDINGS] = df[TableField.EMBEDDINGS].apply(ast.literal_eval)
+        df[TableField.METADATA] = df[TableField.METADATA].apply(ast.literal_eval)
+
+        metadata = df[TableField.METADATA].tolist()
+        embeddings = df[TableField.EMBEDDINGS].tolist()
+        texts = df[TableField.CONTENT].tolist()
+        ids = [str(uuid.uuid1()) for _ in range(len(df))] \
+            if TableField.ID not in df.columns else df[TableField.ID].tolist()
+
+        vectorstore.add_embeddings(
+            texts=texts,
+            embeddings=embeddings,
+            metadatas=metadata,
+            ids=ids
+        )
+        return vectorstore
+
+    @staticmethod
+    def _fetch_data_from_db(settings: VectorStoreConfig) -> pd.DataFrame:
+        """
```

Comment:
should the data extraction be outside of vectorstore loader module?
besides db we could have pdf/url source and maybe data could be extracted somewhere in retrievers and  be passed to vectorstore loader as dataframe

Reply:
This is used to load existing pgvector db that was created outside of langchain